### ISSUE TITLE
proposePointsByInfillOptimization gives unclear error message when getOptStateModels fails

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlrMBO#207]
When getOptStateModels in [this line](https://github.com/mlr-org/mlrMBO/blob/7ca9ad456e18024f8d1d1cacfe8f1a3ad3a67360/R/proposePointsByInfillOptimization.R#L16) fails (which happens e.g. when the learner throws an error), the `coalesce` catches the error and returns `NULL`; at some point later there is an error because `models` is `NULL`, but this is a pain to debug.

**Comment by jakob-r** [comment]
I already noticed it and didn't spot where the error gets lost. Thanks a lot for spotting it. Will investigate to find a better handling. Suggestions?

**Comment by mb706** [comment]
How about the old-fashioned
```R
if (is.null(models)) {
  models = getOptStateModels(opt.state)$models
}
```
? Maybe one would also need to check for `FailureModel` since the learner's `$config` (or the global mlr options) might be set to ignore errors. (This should probably happen inside `trainModel`.) This would be easier if mlr offered a `setLearnerOptions`.

As a side thought, this is one of the points where a bug that is not mlrMBO's fault could crash the optimization process (e.g. unpredictable bad interplay of a learner and a preprocessing / feature selection wrapper), so long-term one might think about giving the user an option to specify a fallback model, or to force an opt.state-save, or to propose a few random points. The random points could be dangerous though, maybe there should be some limit to how many random points are proposed in a row since the user might be disappointed to find that his week long optimization run used the "trivial model".

**Comment by jakob-r** [comment]
Was fixed in #209
